### ISSUE TITLE
show=False raises NameError: name 'plt' is not defined for waterfall plot.

### ISSUE THREAD

**Issue by MaksimEkin** [shap/shap#2520]
Below code for when using ```show=False``` on waterfall plot, it raises **NameError: name 'plt' is not defined** error. 

Reproduce as below:
```python
import xgboost
import shap

X, y = shap.datasets.boston()
model = xgboost.XGBRegressor().fit(X, y)

explainer = shap.Explainer(model)
shap_values = explainer(X)

shap.plots.waterfall(shap_values[0], show=False)
```

**Comment by JulesCollenne** [comment]
Duplicate of #2471
It has been fixed, try updating your shap version.

**Comment by MaksimEkin** [comment]
Thank you! Looks like it has not been fixed in latest [v0.40.0](https://github.com/slundberg/shap/releases/tag/v0.40.0), but installing from the master branch do fixes it.
